### ISSUE TITLE
Bug: [@typescript-eslint/no-unused-vars] Error "is defined but never used" for decorators

### ISSUE THREAD

**Issue by iKBAHT** [typescript-eslint/typescript-eslint#5745]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

---

### Repro Code

```typescript
import { makeObservable, observable } from 'mobx';

export class ConfigStore {
    @observable features = {};
    constructor() {
        makeObservable(this);
    }
}
```

### ESLint Config

```javascript
module.exports = {
    parser: '@typescript-eslint/parser',
    parserOptions: {
        project: './tsconfig.json',
        ecmaFeatures: {
            modules: true,
            jsx: true,
        },
        warnOnUnsupportedTypeScriptVersion: false,
        createDefaultProgram: true,
    },
    globals: {
        // eslint-disable-next-line @typescript-eslint/naming-convention
        __ENVIRONMENT__: 'readonly',
    },
    env: {
        browser: true,
        node: true,
        es6: true,
        jest: true,
    },
    extends: [
        'plugin:@typescript-eslint/recommended',
    ],
    plugins: ['@typescript-eslint', 'react-hooks'],
    settings: {
        react: {
            version: 'detect',
        },
    },
    rules: {
        '@typescript-eslint/ban-types': [
            'error',
            {
                types: {
                    Object: {
                        message:
                            'Avoid using the `Object` type. Did you mean `object`?',
                        fixWith: 'object',
                    },
                    Function: {
                        message:
                            'Avoid using the `Function` type. Prefer a specific function type, like `() => void`.',
                        fixWith: '() => void',
                    },
                    Boolean: {
                        message:
                            'Avoid using the `Boolean` type. Did you mean `boolean`?',
                        fixWith: 'boolean',
                    },
                    Number: {
                        message:
                            'Avoid using the `Number` type. Did you mean `number`?',
                        fixWith: 'number',
                    },
                    String: {
                        message:
                            'Avoid using the `String` type. Did you mean `string`?',
                        fixWith: 'string',
                    },
                    Symbol: {
                        message:
                            'Avoid using the `Symbol` type. Did you mean `symbol`?}}',
                        fixWith: 'symbol',
                    },
                },
            },
        ],
        'react-hooks/exhaustive-deps': 'error',
        'react-hooks/rules-of-hooks': 'error',
        quotes: ['error', 'single'],
        '@typescript-eslint/semi': 'error',
        'no-nested-ternary': 'error',
        'object-curly-spacing': ['error', 'always'],
        'array-bracket-spacing': ['error', 'never'],
        'no-trailing-spaces': 'error',
        'comma-dangle': ['error', 'always-multiline'],
        'eol-last': 'error',
        'no-multiple-empty-lines': ['error', { 'max': 1, 'maxEOF': 0 }],
        'no-unreachable': 'error',
        '@typescript-eslint/no-unused-vars': 'error',
    },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "baseUrl": "./src",
    "experimentalDecorators": true,
    "useDefineForClassFields": true
  },
  "include": [
    "src"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

### Expected Result

No errors when import used decorators

### Actual Result

```
1:26  error  'observable' is defined but never used  @typescript-eslint/no-unused-vars
```
`observable` is MobX decorator
```
    "mobx": "6.6.1",
    "mobx-react": "7.5.2",
```

### Additional Info

Same as https://github.com/typescript-eslint/typescript-eslint/issues/5571

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.6.0` + `5.35.1` + `5.39.0` |
| `@typescript-eslint/parser`            | `5.6.0` + `5.35.1` + `5.39.0` |
| `TypeScript`                           | `4.8.4` |
| `ESLint`                               | `8.13.0` + `8.24.0` | 
| `node`                                 | `14.17.0` + `18.7.0` |

**Comment by bradzacher** [comment]
Why did you skip the playground link field?
The template exists for a reason - for you to provide information to us to help us reproduce your report.

I tried [reproducing this in the playground and it worked fine.](https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=JYWwDg9gTgLgBAbziAhgawKYHkBGBnDKANxRwBsMAaOCfQk8jOAXzgDMoIQ4ByEWgB48A3AChRGAZFhwAxmRR48cAMIQAdm2ABzAMoxoTBKLhwAArQLFSFdhhQwArlAzKAvImZjTsjXhhQjrIGUAAUAJSIJqbI6Nh01oyhMAAWwHjh3iyizEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjoAHoizJ46IngCGqACL9YmNXl3Cwo9FIC+IM0A)
Closing this as no repro. If you can create an isolated repro, happy to look further.

**Comment by iKBAHT** [comment]
For history: problem was solved by `rm -rf node_modules` and `yarn`
